File: app/src/main/java/com/example/dagger2demo/dagger/AppComponent.java
package com.example.dagger2demo.dagger;

import com.example.dagger2demo.dagger.cars.Driver;
import com.example.dagger2demo.dagger.modules.DriverModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = DriverModule.class)
public interface AppComponent {
    Driver getDriver();
}

File: app/src/main/java/com/example/dagger2demo/dagger/AppComponent.kt
package com.example.dagger2demo.dagger

import com.example.dagger2demo.dagger.cars.ActivityComponent
import com.example.dagger2demo.dagger.modules.DieselEngineModule
import com.example.dagger2demo.dagger.modules.DriverModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [DriverModule::class])
interface AppComponent {
    fun getActivityComponent(dieselEngineModule: DieselEngineModule?): ActivityComponent?
}